### ISSUE TITLE
OpenStack subscription disappears during the creation process 

### ISSUE THREAD

**Issue by kolodziejczykm** [servicecatalog/oscm#83]
[Build]
17.6 2018/02/01

[Steps to reproduce]
1.	Go to marketplace 
2.	Subscribe to OpenStack service 5 times in a row 
3.	Go to account “Subscriptions” view and check if 5 subscriptions appear as available to administrate 

 [Expected result]
5 subscriptions of OpenStack should be available

 [Observed result]
3 of 5 subscriptions are created. The 4th and 5th subscription disappears from controller view, instances are not created and status of the 2 subscriptions is “pending”. 

[Details]
After a 30 min i tried subscribing to another OpensStack service and it also failed to create it.  So it might seems like there is some kind of a limit at 3?
I've also gone through the documentation but didn't find anything related to this topic. 

Whole process in screens: 

![1](https://user-images.githubusercontent.com/34572039/35798753-0c755c7e-0a64-11e8-8c58-d2598f67d6c9.JPG)
![3 skrin](https://user-images.githubusercontent.com/34572039/35798789-28bebc4a-0a64-11e8-8fca-ffe9522256a5.JPG)
![4 skrin - 2 gone](https://user-images.githubusercontent.com/34572039/35798791-2a3be05c-0a64-11e8-9702-350e28626812.JPG)
![ctmg - jak wyglada po 5 subskrybcjach](https://user-images.githubusercontent.com/34572039/35798828-428cadda-0a64-11e8-8e78-3713da9f17f5.JPG)
![dashboard z 3](https://user-images.githubusercontent.com/34572039/35798837-4b792e6e-0a64-11e8-9b0c-4acb1a548db8.JPG)

[Logs]
[02/05 10:28:40] [EjbTimerPool - 93] WARN  Mail could not be sent. [Cause: Send failure (javax.mail.AuthenticationFailedException: null)]
[02/05 10:28:40] [EjbTimerPool - 93] ERROR Failure during error mail notification for service instance 'stack-1083b203-ee88-4d17-9e24-aa2d4c920297' with message 'Mail could not be sent. [Cause: Send failure (javax.mail.AuthenticationFailedException: null)]'
[02/05 12:07:57] [EjbTimerPool - 99] WARN  'READY_TIMEOUT' is not set and therefore ignored
[02/05 12:08:12] [EjbTimerPool - 93] WARN  'READY_TIMEOUT' is not set and therefore ignored
[02/05 12:08:12] [EjbTimerPool - 93] WARN  OpenStack platform reported error
org.oscm.app.v2_0.exceptions.AbortException: The creation of the requested resources failed. The service provider has been informed about the issue.
        at org.oscm.app.openstack.controller.Dispatcher.dispatch(Dispatcher.java:333)
        at org.oscm.app.openstack.controller.OpenStackController.getInstanceStatus(OpenStackController.java:240)
        at sun.reflect.GeneratedMethodAccessor484.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.openejb.core.interceptor.ReflectionInvocationContext$Invocation.invoke(ReflectionInvocationContext.java:205)
        at org.apache.openejb.core.interceptor.ReflectionInvocationContext.proceed(ReflectionInvocationContext.java:186)
        at org.apache.openejb.monitoring.StatsInterceptor.record(StatsInterceptor.java:181)
        at org.apache.openejb.monitoring.StatsInterceptor.invoke(StatsInterceptor.java:100)
        at sun.reflect.GeneratedMethodAccessor117.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.openejb.core.interceptor.ReflectionInvocationContext$Invocation.invoke(ReflectionInvocationContext.java:205)
        at org.apache.openejb.core.interceptor.ReflectionInvocationContext.proceed(ReflectionInvocationContext.java:186)
        at org.apache.openejb.core.interceptor.InterceptorStack.invoke(InterceptorStack.java:85)
        at org.apache.openejb.core.stateless.StatelessContainer._invoke(StatelessContainer.java:252)
        at org.apache.openejb.core.stateless.StatelessContainer.invoke(StatelessContainer.java:212)
        at org.apache.openejb.core.ivm.EjbObjectProxyHandler.synchronizedBusinessMethod(EjbObjectProxyHandler.java:265)
        at org.apache.openejb.core.ivm.EjbObjectProxyHandler.businessMethod(EjbObjectProxyHandler.java:260)
        at org.apache.openejb.core.ivm.EjbObjectProxyHandler._invoke(EjbObjectProxyHandler.java:89)
        at org.apache.openejb.core.ivm.BaseEjbProxyHandler.invoke(BaseEjbProxyHandler.java:347)
        at com.sun.proxy.$Proxy206.getInstanceStatus(Unknown Source)
        at org.oscm.app.adapter.APPlatformControllerAdapter.getInstanceStatus(APPlatformControllerAdapter.java:67)
        at org.oscm.app.v2_0.service.APPTimerServiceBean.doHandleControllerProvisioning(APPTimerServiceBean.java:299)
        at org.oscm.app.v2_0.service.APPTimerServiceBean.doHandleSystems(APPTimerServiceBean.java:281)
        at org.oscm.app.v2_0.service.APPTimerServiceBean.handleTimer(APPTimerServiceBean.java:216)
        at sun.reflect.GeneratedMethodAccessor485.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.openejb.core.interceptor.ReflectionInvocationContext$Invocation.invoke(ReflectionInvocationContext.java:205)
        at org.apache.openejb.core.interceptor.ReflectionInvocationContext.proceed(ReflectionInvocationContext.java:186)
        at org.apache.openejb.monitoring.StatsInterceptor.record(StatsInterceptor.java:181)
        at org.apache.openejb.monitoring.StatsInterceptor.AroundTimeout(StatsInterceptor.java:145)
        at sun.reflect.GeneratedMethodAccessor110.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.openejb.core.interceptor.ReflectionInvocationContext$Invocation.invoke(ReflectionInvocationContext.java:205)
        at org.apache.openejb.core.interceptor.ReflectionInvocationContext.proceed(ReflectionInvocationContext.java:186)
        at org.apache.openejb.core.interceptor.InterceptorStack.invoke(InterceptorStack.java:85)
        at org.apache.openejb.core.singleton.SingletonContainer._invoke(SingletonContainer.java:272)
        at org.apache.openejb.core.singleton.SingletonContainer.invoke(SingletonContainer.java:221)
        at org.apache.openejb.core.timer.EjbTimerServiceImpl.ejbTimeout(EjbTimerServiceImpl.java:800)
        at org.apache.openejb.core.timer.EjbTimeoutJob.execute(EjbTimeoutJob.java:39)
        at org.apache.openejb.quartz.core.JobRunShell.run(JobRunShell.java:202)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
[02/05 12:08:12] [EjbTimerPool - 93] WARN  Failure during processing for service instance 'stack-859516a1-0605-4ea1-aacc-1438e291cdc7' with message 'The creation of the requested resources failed. The service provider has been informed about the issue.'
[02/05 12:08:13] [EjbTimerPool - 93] WARN  Mail could not be sent. [Cause: Send failure (javax.mail.AuthenticationFailedException: null)]
[02/05 12:08:13] [EjbTimerPool - 93] ERROR Failure during error mail notification for service instance 'stack-859516a1-0605-4ea1-aacc-1438e291cdc7' with message 'Mail could not be sent. [Cause: Send failure (javax.mail.AuthenticationFailedException: null)]'

**Comment by badziakp** [comment]
There is no limitation of subscription number but Openstack could return some errors and that is why subscriptions disappeared. Maybe there was not enough disc space to create 5 subscriptions. In attached logs there is problem with mail configuration. Please make sure that email address is entered correctly and if email server is configured properly.

**Comment by kolodziejczykm** [comment]
I have retested the issue. Basically what happens is that I can't create more than 3 instances of OpenStack. I tried to create 5 of them again as in the reported bug - 3 of them were created and 2 of them just disappeared. 
The overview looks fine: 
![openstackoverview](https://user-images.githubusercontent.com/34572039/35975329-a384d03e-0cdc-11e8-9c1f-02f1a07e015d.JPG)

And the logs:

[02/08 13:24:21] [EjbTimerPool - 2] WARN  Failure during processing for service instance 'stack-af348fea-b890-48a7-8236-e04abb179094' with message 'The creation of the requested resources failed. The service provider has been informed about the issue.'
[02/08 13:24:21] [EjbTimerPool - 2] WARN  'READY_TIMEOUT' is not set and therefore ignored
[02/08 13:24:22] [EjbTimerPool - 2] WARN  OpenStack platform reported error
org.oscm.app.v2_0.exceptions.AbortException: The creation of the requested resources failed. The service provider has been informed about the issue.
        at org.oscm.app.openstack.controller.Dispatcher.dispatch(Dispatcher.java:333)
        at org.oscm.app.openstack.controller.OpenStackController.getInstanceStatus(OpenStackController.java:240)
        at sun.reflect.GeneratedMethodAccessor391.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.openejb.core.interceptor.ReflectionInvocationContext$Invocation.invoke(ReflectionInvocationContext.java:205)
        at org.apache.openejb.core.interceptor.ReflectionInvocationContext.proceed(ReflectionInvocationContext.java:186)
        at org.apache.openejb.monitoring.StatsInterceptor.record(StatsInterceptor.java:181)
        at org.apache.openejb.monitoring.StatsInterceptor.invoke(StatsInterceptor.java:100)
        at sun.reflect.GeneratedMethodAccessor117.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.openejb.core.interceptor.ReflectionInvocationContext$Invocation.invoke(ReflectionInvocationContext.java:205)
        at org.apache.openejb.core.interceptor.ReflectionInvocationContext.proceed(ReflectionInvocationContext.java:186)
        at org.apache.openejb.core.interceptor.InterceptorStack.invoke(InterceptorStack.java:85)
        at org.apache.openejb.core.stateless.StatelessContainer._invoke(StatelessContainer.java:252)
        at org.apache.openejb.core.stateless.StatelessContainer.invoke(StatelessContainer.java:212)
        at org.apache.openejb.core.ivm.EjbObjectProxyHandler.synchronizedBusinessMethod(EjbObjectProxyHandler.java:265)
        at org.apache.openejb.core.ivm.EjbObjectProxyHandler.businessMethod(EjbObjectProxyHandler.java:260)
        at org.apache.openejb.core.ivm.EjbObjectProxyHandler._invoke(EjbObjectProxyHandler.java:89)
        at org.apache.openejb.core.ivm.BaseEjbProxyHandler.invoke(BaseEjbProxyHandler.java:347)
        at com.sun.proxy.$Proxy206.getInstanceStatus(Unknown Source)
        at org.oscm.app.adapter.APPlatformControllerAdapter.getInstanceStatus(APPlatformControllerAdapter.java:67)
        at org.oscm.app.v2_0.service.APPTimerServiceBean.doHandleControllerProvisioning(APPTimerServiceBean.java:299)
        at org.oscm.app.v2_0.service.APPTimerServiceBean.doHandleSystems(APPTimerServiceBean.java:281)
        at org.oscm.app.v2_0.service.APPTimerServiceBean.handleTimer(APPTimerServiceBean.java:216)
        at sun.reflect.GeneratedMethodAccessor387.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.openejb.core.interceptor.ReflectionInvocationContext$Invocation.invoke(ReflectionInvocationContext.java:205)
        at org.apache.openejb.core.interceptor.ReflectionInvocationContext.proceed(ReflectionInvocationContext.java:186)
        at org.apache.openejb.monitoring.StatsInterceptor.record(StatsInterceptor.java:181)
        at org.apache.openejb.monitoring.StatsInterceptor.AroundTimeout(StatsInterceptor.java:145)
        at sun.reflect.GeneratedMethodAccessor110.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.openejb.core.interceptor.ReflectionInvocationContext$Invocation.invoke(ReflectionInvocationContext.java:205)
        at org.apache.openejb.core.interceptor.ReflectionInvocationContext.proceed(ReflectionInvocationContext.java:186)
        at org.apache.openejb.core.interceptor.InterceptorStack.invoke(InterceptorStack.java:85)
        at org.apache.openejb.core.singleton.SingletonContainer._invoke(SingletonContainer.java:272)
        at org.apache.openejb.core.singleton.SingletonContainer.invoke(SingletonContainer.java:221)
        at org.apache.openejb.core.timer.EjbTimerServiceImpl.ejbTimeout(EjbTimerServiceImpl.java:800)
        at org.apache.openejb.core.timer.EjbTimeoutJob.execute(EjbTimeoutJob.java:39)
        at org.apache.openejb.quartz.core.JobRunShell.run(JobRunShell.java:202)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
[02/08 13:24:22] [EjbTimerPool - 2] WARN  Failure during processing for service instance 'stack-880e8d83-d5c7-49de-a64d-2db4a993329a' with message 'The creation of the requested resources failed. The service provider has been informed about the issue.'

**Comment by kowalczyka** [comment]
Clould you please check the heat logs directly on openstack environment you use for tests?

You should find them in /var/log/heat/ directory of the host on which heat runs.

**Comment by kolodziejczykm** [comment]
@kowalczyka I have faced some problems during the OpenStack subscription creation process when I tried to recreate it. I'll investigate it further because it might be some configuration problems and I'll let you know what I have found.

**Comment by kowalczyka** [comment]
I found the reason for the issue:

heat template which is used by our service creates volume which is attached to every instance. The size of the single volume is 3GB.

![template](https://user-images.githubusercontent.com/14541935/36484524-e746a03a-1719-11e8-82b1-421ea2165e8f.jpg)

The volume group in this case is configured with capacity of 10 GB - after succesful creation of 3 instances, the next one is failed as there is no more space for creating related volume.

![volumes](https://user-images.githubusercontent.com/14541935/36484635-2ef47718-171a-11e8-8797-1040d9ad44e4.jpg)
![lmdriver-1](https://user-images.githubusercontent.com/14541935/36484639-30c5d690-171a-11e8-93b8-38e5268457b7.jpg)

Closing as invalid.
